barretenberg/cpp/src/barretenberg/dsl/acir_format/serde/acir.hpp

```diff
@@ -1,1199 +1,1168 @@
+            static Acir bincodeDeserialize(std::vector<uint8_t>);
+        };
 
-    friend bool operator==(const ExpressionWidth&, const ExpressionWidth&);
-    std::vector<uint8_t> bincodeSerialize() const;
-    static ExpressionWidth bincodeDeserialize(std::vector<uint8_t>);
-};
+        struct Brillig {
+            uint64_t acir_index;
+            uint64_t brillig_index;
 
-struct OpcodeLocation {
+            friend bool operator==(const Brillig&, const Brillig&);
+            std::vector<uint8_t> bincodeSerialize() const;
+            static Brillig bincodeDeserialize(std::vector<uint8_t>);
+        };
 
-    struct Acir {
-        uint64_t value;
+        std::variant<Acir, Brillig> value;
 
-        friend bool operator==(const Acir&, const Acir&);
+        friend bool operator==(const OpcodeLocation&, const OpcodeLocation&);
         std::vector<uint8_t> bincodeSerialize() const;
-        static Acir bincodeDeserialize(std::vector<uint8_t>);
+        static OpcodeLocation bincodeDeserialize(std::vector<uint8_t>);
     };
 
-    struct Brillig {
-        uint64_t acir_index;
-        uint64_t brillig_index;
+    struct PublicInputs {
+        std::vector<Circuit::Witness> value;
 
-        friend bool operator==(const Brillig&, const Brillig&);
+        friend bool operator==(const PublicInputs&, const PublicInputs&);
         std::vector<uint8_t> bincodeSerialize() const;
-        static Brillig bincodeDeserialize(std::vector<uint8_t>);
+        static PublicInputs bincodeDeserialize(std::vector<uint8_t>);
     };
 
-    std::variant<Acir, Brillig> value;
-
-    friend bool operator==(const OpcodeLocation&, const OpcodeLocation&);
-    std::vector<uint8_t> bincodeSerialize() const;
-    static OpcodeLocation bincodeDeserialize(std::vector<uint8_t>);
-};
-
-struct PublicInputs {
-    std::vector<Circuit::Witness> value;
-
-    friend bool operator==(const PublicInputs&, const PublicInputs&);
-    std::vector<uint8_t> bincodeSerialize() const;
-    static PublicInputs bincodeDeserialize(std::vector<uint8_t>);
-};
-
-struct Circuit {
-    uint32_t current_witness_index;
-    std::vector<Opcode> opcodes;
-    ExpressionWidth expression_width;
-    std::vector<Witness> private_parameters;
-    PublicInputs public_parameters;
-    PublicInputs return_values;
-    std::vector<std::tuple<OpcodeLocation, std::string>> assert_messages;
-    bool recursive;
-
-    friend bool operator==(const Circuit&, const Circuit&);
-    std::vector<uint8_t> bincodeSerialize() const;
-    static Circuit bincodeDeserialize(std::vector<uint8_t>);
-};
+    struct Circuit {
+        uint32_t current_witness_index;
+        std::vector<Circuit::Opcode> opcodes;
+        Circuit::ExpressionWidth expression_width;
+        std::vector<Circuit::Witness> private_parameters;
+        Circuit::PublicInputs public_parameters;
+        Circuit::PublicInputs return_values;
+        std::vector<std::tuple<Circuit::OpcodeLocation, std::string>> assert_messages;
```

Comment:
I think this changes need to be removed! Also it's missing passing the formatter on the whole file

Reply:
To clarify, it should look like the previous diff without the `Circuit::` prefixes.

Reply:
Hmm... I thought this would have the same contents as the generated `acir.cpp` with the previously mentioned replacement only. Do you have the procedure to update this file documented somewhere?

Reply:
@ggiraldez unfortunately the documentation on this process is lacking. The replacement mentioned here is the only other place. I'm going to quickly do the serialization for you and also make an issue to document this process.

Reply:
Ok the committed changes look correct. Let's see what CI says. I will document the process for any future updates